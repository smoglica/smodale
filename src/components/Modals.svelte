<script>
  import Modal from './Modal.svelte';
  import modals from '../store';
</script>

<div class="js-modals">
  {#each $modals.dynamic as modal (modal.props.name)}
    <Modal
      {...modal.props}
      on:opened={modal.events && modal.events.opened}
      on:closed={modal.events && modal.events.closed}
    >
      <svelte:component this={modal.component} {...modal.componentProps} />
    </Modal>
  {/each}
</div>
