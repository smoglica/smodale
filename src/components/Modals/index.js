import Modals from './Modals.svelte';

export default Modals;
