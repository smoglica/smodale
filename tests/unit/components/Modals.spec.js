import { render } from '@testing-library/svelte';
import Modals from 'smodale/components/Modals';

describe('Modals', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(Modals);
  });

  it('should instantiate correctly', () => {
    expect(wrapper.container).toMatchSnapshot();
  });
});
