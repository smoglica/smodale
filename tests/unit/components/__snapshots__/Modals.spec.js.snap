// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals should instantiate correctly 1`] = `
<body>
  <div>
    <div
      class="js-modals"
    />
  </div>
</body>
`;
